runtime/gc_vlhgc/SchedulingDelegate.cpp

```diff
@@ -524,44 +663,195 @@ MM_SchedulingDelegate::calculateEstimatedGlobalBytesToScan() const
+{
+	double predictedNumberOfCollections = predictNumberOfCollections(env, currentEdenSize, edenSizeChange, freeTenure);
+	double predictedIntervalTime = predictIntervalBetweenCollections(env, currentEdenSize, edenSizeChange, pgcAvgIntervalTime);
+	double predictedAvgPgcTime = predictPgcTime(env, currentEdenSize, edenSizeChange);
+	
+	uint64_t gmpTime = _totalGMPWorkTimeUs;
+	if (0 == gmpTime) {
+		/* GMP has not yet happened, so make a rough guess - but a relatively high guess, so that eden thinks GMP is very expensive relative to PGC */
+		gmpTime = 20 * (_historicalPartialGCTime * 1000);
+	}
+
+	double gcActiveTime = (double)gmpTime + (predictedAvgPgcTime * predictedNumberOfCollections);
+	double totalIntervalTime = (double)gmpTime + ((predictedAvgPgcTime + predictedIntervalTime) * predictedNumberOfCollections);
+
+	double estimatedOverhead = gcActiveTime / totalIntervalTime;
+	return estimatedOverhead;
+}
+
+double 
+MM_SchedulingDelegate::predictIntervalBetweenCollections(MM_EnvironmentVLHGC *env, uintptr_t currentEdenSize, intptr_t edenSizeChange, uint64_t pgcAvgIntervalTime)
+{	
+	/* The interval between PGC collections is proportional to eden size. Ex. If eden size doubles, we expect the interval between PGC collections to double as well */
+	double intervalChange = (double)(currentEdenSize + edenSizeChange)/currentEdenSize;
+	return (double)pgcAvgIntervalTime * intervalChange;
+}
+
+double 
+MM_SchedulingDelegate::predictNumberOfCollections(MM_EnvironmentVLHGC *env, uintptr_t currentEdenSize, intptr_t edenSizeChange, uintptr_t freeTenure)
+{
+	/* The number of PGC collections is proportional to how much free tenure will be left after we expand/contract eden */
+	double collectionCountChange = (double)(freeTenure - edenSizeChange)/freeTenure;
+	return (double)_extensions->globalVLHGCStats.getRepresentativePgcPerGmpCount() * collectionCountChange;
+}
+
+double
+MM_SchedulingDelegate::predictPgcTime(MM_EnvironmentVLHGC *env, uintptr_t currentEdenSize, intptr_t edenSizeChange) 
+{	
+	/* 
+	 * PGC avg time MAY be related to eden size. Certain applications/allocation patterns, will cause pgc time to increase as eden increases,
+	 * while certain different workloads may keep pgc time relatively constant even as eden size increases. 
+	 * Create a model to determine how pgc time will be afffected by eden size - keeping in mind that _pgcTimeIncreasePerEdenFactor can vary depending on the application
+	 */
+	double edenSizeGb = (double)currentEdenSize / 1000000000;
+	double edenChangeGb = (double)edenSizeChange/ 1000000000;
+	double edenChangeRatio = (edenChangeGb + edenSizeGb + 1.0) / (edenSizeGb + 1.0);
+
+	/* Use a math workaround for "log base _pgcTimeIncreasePerEdenFactor (edenChangeRatio) "*/
+	double pgcTimeChangeForEdenChange = log(edenChangeRatio) / log(_pgcTimeIncreasePerEdenFactor);
```

Comment:
This variable name sounds "heavy". `pgcTimeChangeForEden`? `pgcTimeDeltaForEdenChange`? Something else?

Reply:
Agreed. I like `pgcTimeDeltaForEdenChange`